backend/mpmediabridge.m

```diff
@@ -0,0 +1,85 @@
+//go:build darwin
+
+#import "mpmediabridge.h"
+
+/**
+ * C bridge registering callbacks for media playback events using the native CommandCenter API.
+ */
+void register_os_remote_commands() {
+    MPRemoteCommandCenter *commandCenter = [MPRemoteCommandCenter sharedCommandCenter];
+    [commandCenter.playCommand addTargetWithHandler:^MPRemoteCommandHandlerStatus(MPRemoteCommandEvent * _Nonnull event) {
+        os_remote_command_callback(PLAY);
+        return MPRemoteCommandHandlerStatusSuccess;
+    }];
+
+    [commandCenter.pauseCommand addTargetWithHandler:^MPRemoteCommandHandlerStatus(MPRemoteCommandEvent * _Nonnull event) {
+        os_remote_command_callback(PAUSE);
+        return MPRemoteCommandHandlerStatusSuccess;
+    }];
+
+    [commandCenter.togglePlayPauseCommand addTargetWithHandler:^MPRemoteCommandHandlerStatus(MPRemoteCommandEvent * _Nonnull event) {
+        os_remote_command_callback(TOGGLE);
+        return MPRemoteCommandHandlerStatusSuccess;
+    }];
+
+    [commandCenter.stopCommand addTargetWithHandler:^MPRemoteCommandHandlerStatus(MPRemoteCommandEvent * _Nonnull event) {
+        os_remote_command_callback(STOP);
+        return MPRemoteCommandHandlerStatusSuccess;
+    }];
+
+    [commandCenter.nextTrackCommand addTargetWithHandler:^MPRemoteCommandHandlerStatus(MPRemoteCommandEvent * _Nonnull event) {
+        os_remote_command_callback(NEXT_TRACK);
+        return MPRemoteCommandHandlerStatusSuccess;
+    }];
+
+    [commandCenter.previousTrackCommand addTargetWithHandler:^MPRemoteCommandHandlerStatus(MPRemoteCommandEvent * _Nonnull event) {
+        os_remote_command_callback(PREVIOUS_TRACK);
+        return MPRemoteCommandHandlerStatusSuccess;
+    }];
+}
+
+/**
+ * C bridge setting "Now Playing" information on macOS for media playback using the native APIs.
+ */
+void set_os_now_playing_info(const char *title, const char *artist, const char *coverArtFileURL) {
+    NSString *coverArtLocationString = [NSString stringWithUTF8String:coverArtFileURL];
+    NSURL *coverArtURL = [NSURL URLWithString:coverArtLocationString];
+    NSImage *coverArtImage = [[NSImage alloc] initWithContentsOfURL:coverArtURL];
```

Comment:
I don't know how memory management works in Obj-C, but will the images be garbage collected or is a manual free needed?

Reply:
Good question. How this works is; in Objective-C we can rely on ARC (automated reference counting) to take care of the memory management for these objects and so there should be no need to release them manaually.